internal/yunikorn/sync.go

```diff
@@ -9,17 +9,18 @@ import (
 	"github.com/G-Research/yunikorn-core/pkg/webservice/dao"
 	"github.com/oklog/ulid/v2"
 
-	"github.com/G-Research/unicorn-history-server/internal/log"
 	"github.com/G-Research/unicorn-history-server/internal/model"
 	"github.com/G-Research/unicorn-history-server/internal/util"
 )
 
 // syncPartitions fetches partitions from the Yunikorn API and syncs them into the database
-func (s *Service) syncPartitions(ctx context.Context) ([]*model.Partition, error) {
-	// Get partitions from Yunikorn API and upsert into DB
-	partitions, err := s.client.GetPartitions(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("could not get partitions: %v", err)
+func (s *Service) syncPartitions(ctx context.Context, partitions []*dao.PartitionInfo) error {
+	if partitions == nil {
+		var err error
+		partitions, err = s.client.GetPartitions(ctx)
+		if err != nil {
+			return fmt.Errorf("could not get partitions: %w", err)
+		}
```

Comment:
Why would we fetch partitions if they are not passed in?
I'd just do `if len(partitions) == 0 { return nil }`

Reply:
`syncPartitions()` is also used in the accumulator, because we do not have events for partition in the event stream. 
ref: https://github.com/G-Research/unicorn-history-server/blob/545795dfc935636c1c0b9a4a9a80e7105f7e269e/internal/yunikorn/service.go#L43-L50

Reply:
Oh that part should be removed completely.
When we sync queues, we should check if partition exists and sync partitions if not

Reply:
Agree - syncing partitions on every EventRecord received would be expensive.

Reply:
If we only sync when the partition does not exist, we cannot handle 'partition update' scenarios. I believe we should sync the partition for all queue events, regardless of whether the partition already exists.